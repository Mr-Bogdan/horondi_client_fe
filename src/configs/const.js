export const AUTH_ERRORS = {
  ACCESS_TOKEN_IS_NOT_VALID: 'USER_NOT_AUTHORIZED',
  REFRESH_TOKEN_IS_NOT_VALID: 'REFRESH_TOKEN_IS_NOT_VALID'
};
export const COURIER = 'COURIER';
export const RESET = 'reset';

export const CURRENCY = {
  UAH: 'UAH',
  USD: 'USD'
};
export const IMG_ALT = {
  REGISTER_IMG_INFO: 'register info'
};

export const languageName = 'language';
export const limitHistoryOrders = 10;

export const TEXT_FIELD_VARIANT = {
  OUTLINED: 'outlined',
  ROUNTED: 'rounded'
};
export const TEXT_FIELD_SIZE = {
  SMALL: 'small'
};
export const MATERIAL_UI_COLOR = {
  INHERIT: 'inherit',
  PRIMARY: 'primary',
  ACTION: 'action'
};
export const SLIDER_HOME_PAGE = {
  ARROW: 'arrow',
  SLIDER: 'slider'
};

export const statusColors = [
  { label: 'CREATED', color: '#9e9e9e' },
  { label: 'CONFIRMED', color: '#4791db' },
  { label: 'PRODUCED', color: '#81c784' },
  { label: 'CANCELLED', color: '#e33371' },
  { label: 'REFUNDED', color: '#ffb74d' },
  { label: 'SENT', color: '#b39ddb' },
  { label: 'DELIVERED', color: '#546e7a' }
];

export const TYPES_CONST = {
  NUMBER: 'number',
  STRING: 'string'
};
