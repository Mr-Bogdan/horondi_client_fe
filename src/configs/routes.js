const routes = {
  pathToLogin: '/login',
  pathToRecovery: '/recovery',
  pathToRegister: '/register',
  pathToProducts: '/product'
};
export default routes;
