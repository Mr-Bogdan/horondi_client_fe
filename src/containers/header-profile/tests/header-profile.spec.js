import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { MenuItem } from '@material-ui/core';
import HeaderProfile from '../header-profile';
import { getFromLocalStorage } from '../../../services/local-storage.service';
import { DARK_THEME, LIGHT_THEME } from '../../../configs';

jest.mock('../header-profile.styles', () => ({ useStyles: () => ({}) }));
jest.mock('../../../services/local-storage.service');
jest.mock('react-redux');
jest.mock('connected-react-router', () => ({ push: () => 0 }));

const props = {
  fromSideBar: '',
  setIsMenuOpen: () => true,
  onClick: jest.fn(),
  onclose: jest.fn(),
  clickHandler: {
    handleClose: () => true,
    handleKeyDown: () => {},
    handleChangeTheme: () => true,
    handleRedirect: () => true
  }
};
const mockStore = {
  lightMode: true,
  userData: false
};
const mockDispatch = jest.fn();
let theme = LIGHT_THEME;

getFromLocalStorage.mockImplementation(() => theme);

useSelector.mockImplementation(() => mockStore);
useDispatch.mockReturnValue(mockDispatch);

describe('<HeaderProfile />', () => {
  it('should render <HeaderProfile />', () => {
    theme = DARK_THEME;
    const wrapper = shallow(<HeaderProfile {...props} />);
    const menu = wrapper.find(`[data-cy='menuItem']`).at(0);
    menu.props().onClick({ target: { setIsMenuOpen: false } });
    expect(wrapper).toBeDefined();
  });
  it('should simulate click on second child of <HeaderProfile />', () => {
    const wrapper = shallow(<HeaderProfile {...props} />);
    const menu = wrapper.find(`[data-cy='menuItem']`).at(1);
    menu.simulate('click', { persist: jest.fn(), setIsMenuOpen: () => true });
    expect(props.setIsMenuOpen()).toEqual(true);
  });
  it('should work with logout user', () => {
    const wrapper = shallow(<HeaderProfile {...props} />);
    const iconOut = wrapper.find(`[data-cy='iconOut']`);
    iconOut.simulate('click', { persist: jest.fn() });
    expect(wrapper).toBeDefined();
  });
  it('should work with login user', () => {
    mockStore.userData = true;
    const wrapper = shallow(<HeaderProfile {...props} />);
    const iconIn = wrapper.find(`[data-cy='iconIn']`);
    iconIn.simulate('click', { persist: jest.fn(), handleChangeTheme: DARK_THEME });
    expect(wrapper.find(MenuItem).length).toEqual(5);
  });
  it('should simulate clicks on children', () => {
    mockStore.userData = true;
    const wrapper = shallow(<HeaderProfile {...props} />);
    const menuItem2 = wrapper.find(`[data-cy='menuItem']`).at(2);
    const menuItem3 = wrapper.find(`[data-cy='menuItem']`).at(3);
    const menuItem4 = wrapper.find(`[data-cy='menuItem']`).at(4);
    menuItem2.simulate('click', { persist: jest.fn(), setIsMenuOpen: false });
    menuItem3.simulate('click', { persist: jest.fn() });
    menuItem4.simulate('click', { persist: jest.fn() });
    expect(wrapper).toBeDefined();
  });
  it('should work onClose', () => {
    const wrapper = shallow(<HeaderProfile {...props} />);
    const menu = wrapper.find(`[data-cy='menu']`);
    menu.props().onClose({ target: { handleClose: true } });
    expect(wrapper.props().children.length).toEqual(2);
  });
});
