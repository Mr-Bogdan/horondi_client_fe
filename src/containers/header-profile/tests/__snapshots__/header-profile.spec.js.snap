// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderProfile /> Should match snapshot 1`] = `
<div
  data-cy="profile"
>
  <Memo(ForwardRef(PersonIcon))
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  />
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    elevation={0}
    getContentAnchorEl={null}
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <WithStyles(ForwardRef(MenuItem))
      disableGutters={true}
      key="Список уподобань"
      onClick={[Function]}
    >
      <Memo(ForwardRef(FavoriteIcon)) />
      Список уподобань
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      disableGutters={true}
      key="Змінити тему"
      onClick={[Function]}
    >
      <Memo(ForwardRef(Brightness7Icon)) />
      Змінити тему
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      disableGutters={true}
      key="Профіль"
      onClick={[Function]}
    >
      <Memo(ForwardRef(PersonOutlineIcon)) />
      Профіль
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      disableGutters={true}
      key="Історія замовлень"
      onClick={[Function]}
    >
      <Memo(ForwardRef(HistoryIcon)) />
      Історія замовлень
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      disableGutters={true}
      key="Вийти"
      onClick={[Function]}
    >
      <Memo(ForwardRef(ExitToAppIcon)) />
      Вийти
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
